rust/agama-dbus-server/src/network/model.rs

```diff
@@ -480,6 +550,47 @@ pub struct LoopbackConnection {
 }
 
 #[derive(Debug, Default, PartialEq, Clone)]
+pub struct BondConnection {
+    pub base: BaseConnection,
+    pub bond: BondConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct BondOptions(pub HashMap<String, String>);
+
+impl TryFrom<String> for BondOptions {
+    type Error = NetworkStateError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let mut options = HashMap::new();
+
+        for opt in value.split_whitespace() {
+            let opt_word: Vec<&str> = opt.trim().split('=').collect();
+            options.insert(opt_word[0].to_string(), opt_word[1].to_string());
+        }
+
+        Ok(BondOptions(options))
+    }
+}
+
+impl fmt::Display for BondOptions {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut opts = vec![];
+        for (key, value) in &self.0 {
```

Comment:
As an exercise, could you try a `map`? IMHO, the best thing of using combinators is that they express an intention.